core/controlplane/config/templates/stack-template.json

```diff
@@ -257,15 +242,37 @@
                   "Resource": "*",
                   "Effect": "Allow"
                 }
+          ]
+        }
+      }
+    },
+    "IAMRoleController": {
+      "Properties": {
+        "AssumeRolePolicyDocument": {
+          "Statement": [
+            {
+              "Action": [
+                "sts:AssumeRole"
               ],
-              "Version": "2012-10-17"
-            },
-            "PolicyName": "root"
-          }
-        ]
+              "Effect": "Allow",
+              "Principal": {
+                "Service": [
+                  "ec2.{{.Region.PublicDomainName}}"
+                ]
+              }
+            }
+          ],
+          "Version": "2012-10-17"
+        },
+        "Path": "/",
+        {{if .Controller.IAMConfig.ManagedIamRoleName }}
+        "RoleName":  {"Fn::Join": ["",[{"Ref": "AWS::Region"},"-","{{.Controller.IAMConfig.ManagedIamRoleName}}"]]},
+        {{end}}
+        "ManagedPolicyArns": [ {"Ref": "IAMManagedPolicyController"} ]
```

Comment:
i don't mind at all, i like `iam.role.name`. Regarding your second point,

between

```
iam:
     role:
           name: "yourRoleName"
           managedPolicyArns:
                - "arn:aws:iam:YOURACCOUNTID/policy/POLICYID"
     instanceProfile:
          arn: "arn:aws:iam::YOURACCOUNTID:instance-profile/INSTANCEPROFILEID"
```
and 
```
iam:
     instanceProfile:
        role:
           name: "yourRoleName"
           managedPolicyArns:
                - "arn:aws:iam:YOURACCOUNTID/policy/POLICYID"
        arn: "arn:aws:iam::YOURACCOUNTID:instance-profile/INSTANCEPROFILEID"
```
 there is a link between the role and the instance profile i like the first slightly more because the option is more clear either you go with a role or either you set an instance profile but you cannot do both. 

That said, both can work, so if you feel that the second makes more sense i'll do that.

Thanks @mumoshu :bowing_man: